src/service/asset-api-adapter.js

```diff
@@ -0,0 +1,248 @@
+        }
+    };
+    var serviceOptions = $.extend(true, {}, defaults, config);
+    var urlConfig = new ConfigService(serviceOptions).get('server');
+
+    if (!serviceOptions.account) {
+        serviceOptions.account = urlConfig.accountPath;
+    }
+
+    if (!serviceOptions.project) {
+        serviceOptions.project = urlConfig.projectPath;
+    }
+
+    var transportOptions = $.extend(true, {}, serviceOptions.transport, {
+        url: urlConfig.getAPIPath(apiEndpoint)
+    });
+
+    if (serviceOptions.token) {
+        transportOptions.headers = {
+            'Authorization': 'Bearer ' + serviceOptions.token
+        };
+    }
+
+    var http = new TransportFactory(transportOptions);
+
+    var assetApiParams = ['encoding', 'data', 'contentType'];
+    var scopeConfig = {
+        user: ['scope', 'account', 'project', 'group', 'userId'],
+        group: ['scope', 'account', 'project', 'group'],
+        project: ['scope', 'account', 'project'],
+    };
+
+    var validateFilename = function (filename) {
+        if (!filename) {
+            throw new Error('filename is needed.');
+        }
+    };
+
+    var validateUrlParams = function (options) {
+        var partKeys = scopeConfig[options.scope];
+        if (!partKeys) {
+            throw new Error('scope parameter is needed.');
+        }
+
+        $.each(partKeys, function () {
+            if (!options[this]) {
+                throw new Error(this + ' parameter is needed.');
+            }
+        });
+    };
+
+    var buildUrl = function (filename, options) {
+        validateUrlParams(options);
+        var partKeys = scopeConfig[options.scope];
+        var parts = $.map(partKeys, function (key) {
+            return options[key];
+        });
+        if (filename) {
+            // This prevents adding a trailing / in the URL as the Asset API
+            // does not work correctly with it
+            filename = '/' + filename;
+        }
+        return urlConfig.getAPIPath(apiEndpoint) + parts.join('/') + filename;
+    };
+
+    var publicAPI = {
+        /**
+        * Private function, all requests follow a more or less same approach to
+        * use the Asset API and the difference is the HTTP verb
+        *
+        * @param {string} `method` (Required) HTTP verb
+        * @param {string} `filename` (Required) Name of the file to delete/replace/create
+        * @param {object} `params` (Optional) Body parameters to send to the Asset API
+        * @param {object} `options` (Optional) Options object to override global options.
+        *
+        */
+        _upload: function (method, filename, params, options) {
+            validateFilename(filename);
+            // make sure the parameter is clean
+            method = method.toLowerCase();
+            var urlOptions = $.extend({}, serviceOptions, options);
+            // whitelist the fields that we actually can send to the api
+            if (urlOptions.contentType === 'application/json') {
```

Comment:
reviewed with @rvillarreal -- this is not picking up the contentType correctly. we think this should be urlOptions.transport.contentType (or something similar that has the same effect)